### PR TITLE
Disable softmax tests on ROCm

### PR DESCRIPTION
**Pull request by ashishfarmer**
This PR disables the test_softmax and test_softmax_results in test_nn.py that were enabled in #46363. The softmax tests are causing failure on gfx906 machines. Disabling those until we root cause and fix them on 906.

cc: @jeffdaily @ezyang

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

**Comment by jeffdaily** [comment]
@ezyang can we merge this today?  The flaky test that this PR will skip is confusing other PRs (https://github.com/pytorch/pytorch/pull/46878, CC @xw285cornell ).

**Review by xw285cornell — APPROVED** [review]
There was some flaky tests internally so we're rebasing it. Should get land soon.
